lib/omniauth/strategies/openid_connect.rb

```diff
@@ -210,6 +214,15 @@ def decode_id_token(id_token)
         ::OpenIDConnect::ResponseObject::IdToken.decode(id_token, public_key)
       end
 
+      def decoded_id_tokens(id_token)
```

Comment:
1) This method does not return `tokens`! You should not name a method like this just because it also can cache multiple values.

2) Why do we need this caching anyway? Do you have any problems with the performance of this method? This code looks like a great example of premature optimization.